### ISSUE TITLE
about segmentation network

### ISSUE THREAD

**Issue by touristCheng** [charlesq34/pointnet#48]
Hi! Thanks for sharing your nice works. I wonder why the segmentation network need to concatenate the one hot label with the pooling feature, will remove the encoded label decrease the performance? Thanks. @charlesq34

**Comment by daerduoCarey** [comment]
The one-hot vector is there for fair comparison to the other baseline methods which target at segmentation for shapes from a given same object category. Our part-seg PointNet can work without one-hot vector but there will be some performance drop, around 1-2%. The fact that our PointNet can train across different object categories leads to both positive effects (e.g. easier to train, no overfitting to small subcategories, can be easily finetuned to new shapes) and negative ones (e.g. some segmentation labels may be from different object categories).

Thank you.

Bests,
Kaichun

**Comment by touristCheng** [comment]
Thanks very much! I want to know where I can get the original 3D models for sampling the points, and how can I visualize the segmentation results, thanks for your help.

**Comment by sitzikbs** [comment]
@touristCheng If you are reffering to the classification it is in the [ModelNet40 ](http://modelnet.cs.princeton.edu/)website. If it is to the segmentation then I think it is in the [shapenet ](https://www.shapenet.org/) ( not sure though). The code has a file to download the sampled data. 

I too will appreciate a visualization code for the segmentation results ( Im sure there is  a file like this somewhere since the paper shows  these results)

**Comment by sheromon** [comment]
I recently started playing with this model. (BTW, thank you @charlesq34 for this excellent release. It is much higher-quality and easier to use than other projects that I've looked at.) @sitzikbs and @touristCheng, there are programs such as MeshLab that will recognize the .obj file format. Using MeshLab, I just imported the "pred" (prediction) and "gt" (ground truth) .obj files separately, and you can compare them. I imagine that other programs for 3D visualization will do the same.

**Comment by sitzikbs** [comment]
Thanks @sheromon. I already wrote some code to visualize it. Hopefully Ill upload a branch to this in a few months.

**Comment by charlesq34** [comment]
seems the issue has been resolved. closing for now.

**Comment by rejivipin** [comment]
@sitzikbs 
Have you written the code for visualization?

**Comment by sitzikbs** [comment]
@rejivipin 
try the functions here : 
https://github.com/sitzikbs/3DmFV-Net/blob/master/utils/visualization.py

**Comment by noridayu1998** [comment]
> @rejivipin
> try the functions here :
> https://github.com/sitzikbs/3DmFV-Net/blob/master/utils/visualization.py

this is for visualizing the semantic segmentation result?

**Comment by sitzikbs** [comment]
The function ` visualize_pc_seg` should work but I am sure that there are better implementations by now.

**Comment by noridayu1998** [comment]
> The function ` visualize_pc_seg` should work but I am sure that there are better implementations by now.

@sitzikbs 
the function visualize_pc_seg will export image file. there is possible to change it in point cloud file?
